src/Microsoft.Health.Fhir.TemplateManagement/OCIFileManager.cs

```diff
@@ -5,90 +5,99 @@
 
 using System.Collections.Generic;
 using System.IO;
-using System.Linq;
 using System.Threading.Tasks;
-using EnsureThat;
+using Microsoft.Azure.ContainerRegistry.Models;
 using Microsoft.Health.Fhir.TemplateManagement.Client;
 using Microsoft.Health.Fhir.TemplateManagement.Models;
 using Microsoft.Health.Fhir.TemplateManagement.Overlay;
+using Newtonsoft.Json;
 
 namespace Microsoft.Health.Fhir.TemplateManagement
 {
     public class OCIFileManager
     {
-        private readonly OrasClient _orasClient;
+        private readonly IOCIClient _client;
         private readonly OverlayFileSystem _overlayFS;
         private readonly OverlayOperator _overlayOperator;
+        private readonly string _imageReference;
 
         public OCIFileManager(string imageReference, string workingFolder)
         {
             ImageInfo.ValidateImageReference(imageReference);
-            _orasClient = new OrasClient(imageReference);
+            _imageReference = imageReference;
+
             _overlayFS = new OverlayFileSystem(workingFolder);
+            _client = new OrasClient(Path.Combine(workingFolder, Constants.HiddenLayersFolder));
             _overlayOperator = new OverlayOperator();
         }
 
-        public async Task PullOCIImageAsync()
+        /// <summary>
+        /// Pull image layers into working folder.
+        /// Layers will be written into hidden image folder.
+        /// </summary>
+        /// <returns>Image information.</returns>
+        public async Task<ImageInfo> PullOCIImageAsync()
         {
             _overlayFS.ClearImageLayerFolder();
-            await _orasClient.PullImageAsync(_overlayFS.WorkingImageLayerFolder);
+            var imageInfo = await _client.PullImageAsync(_imageReference);
+            _overlayFS.WriteManifest(JsonConvert.SerializeObject(imageInfo.Manifest));
```

Comment:
move JsonConvert.SerializeObject to other place?

Reply:
Moved it into OverlayFileSystem.